src/js/jsx/tools/SuperselectOverlay.jsx

```diff
@@ -126,6 +127,21 @@ define(function (require, exports, module) {
             };
         },
 
+        shouldComponentUpdate: function (nextProps, nextState) {
+            var lastLayers = this.state.document.layers.selected,
+                nextLayers = nextState.document.layers.selected,
+                lastLayerProps = collection.pluckAll(lastLayers, ["id", "bounds"]),
+                nextLayerProps = collection.pluckAll(nextLayers, ["id", "bounds"]),
+                layersChanged = !Immutable.is(lastLayerProps, nextLayerProps);
+
+            return layersChanged ||
+                this.state.marqueeEnabled !== nextState.marqueeEnabled ||
```

Comment:
Would it make more sense to do these faster checks first before falling back to the heavier check based on pluck?